doc/design/register_grad_op.md

```diff
@@ -0,0 +1,69 @@
+# Design Doc: Register Gradient Operator
+
+## Problem
+
+Since we separate users program in two stages, compile time and runtime, we should record and look up the mapping relationship between an operator and its gradient operators when compile. However, we register this relationship in runtime by these `OpInfo` fields.
```

Comment:
The execution of a neural network topology in PaddlePaddle is separated into two stages, complie-time and run-time.
At complie-time, a ProgramDesc will be generated. At run-time, the ProgramDesc will be executed on specific hardware. We can refer to the design of [computation-graphs](https://github.com/PaddlePaddle/Paddle/blob/develop/doc/design/refactorization.md#computation-graphs).
The Gradient Operator's OpDesc is also generated in compile-time. We have to find the mapping relationship between Operator's OpDesc and its GradientOp's OpDesc.
However, we make the mapping relationship between Operator and its GradientOp in run-time in `OpInfo` class currently.